### PR TITLE
Improve ANSSI R28

### PR DESCRIPTION
**Pull request by jan-cerny**
#### Description:

Create new rule `systemd_tmp_mount` that checks if the systemd `tmp.mount` mount unit is enabled.

This rule uses a new template that is also added by this PR.

This rule conflicts with the current implementation of the rule `accounts_polyinstantiated_tmp` because rule `accounts_polyinstantiated_tmp` requires the existence of the `/tmp/tmp-inst` directory and this directory is removed by systemd `tmp.mount` mount unit during reboot. We will address this conflict by removal of the requirement for the existence of 
`/tmp/tmp-inst`. We can do this because this directory is created automatically by `pam` if it doesn't exist. We will only check that if the directory exists it has correct mode to prevent fails of the polyinstantiation.

For more details, please read commit messages of all commits.

#### Rationale:

We will add this rule to ANSSI R28 because it aligns with the requirement to clean after reboot or be preferably type tmpfs.

### CODE REVIEW

**Comment by jan-cerny** [comment]
From the CI results we can see that the new rule conflicts with accounts_polyinstantiated_tmp which is also a part of the ANSSI profile.

File: linux_os/guide/system/software/disk_partitioning/systemd_tmp_mount_enabled/rule.yml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+severity: low
+
+identifiers:
+    cce@rhel7: CCE-86889-3
+    cce@rhel8: CCE-86890-1
+    cce@rhel9: CCE-86891-9
+
+ocil: |-
+    {{{ systemd_mount_enabled("tmp") }}}
```
Comment (Mab879) [inline linux_os/guide/system/software/disk_partitioning/systemd_tmp_mount_enabled/rule.yml:25]: This will be need to updated as well.

**Review by Mab879 — CHANGES_REQUESTED** [review]
Just one minor item.

**Comment by jan-cerny** [comment]
I have renamed the OCIL macro.

**Comment by Mab879** [comment]
Automatus passes locally

```
$ ./automatus.py rule --datastream ../build/ssg-rhel8-ds.xml --libvirt qemu:///system automatus_rhel8_10 systemd_tmp_mount_enabled,accounts_polyinstantiated_tmp
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/mburket/review/ComplianceAsCode/content3/tests/logs/rule-custom-2024-03-14-0924/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_accounts_polyinstantiated_tmp
INFO - Script correct.pass.sh using profile (all) OK
INFO - Script line_not_there.fail.sh using profile (all) OK
INFO - Script wrong_mode.fail.sh using profile (all) OK
INFO - Script directory_doesnt_exist.pass.sh using profile (all) OK
INFO - xccdf_org.ssgproject.content_rule_systemd_tmp_mount_enabled
INFO - Script mount_enabled.pass.sh using profile (all) OK
INFO - Script mount_disabled.fail.sh using profile (all) OK
$ ./automatus.py rule --datastream ../build/ssg-rhel8-ds.xml --libvirt qemu:///system automatus_rhel8_10 systemd_tmp_mount_enabled,accounts_polyinstantiated_tmp
Setting console output to log level INFO
INFO - The base image option has not been specified, choosing libvirt-based test environment.
INFO - Logging into /home/mburket/review/ComplianceAsCode/content3/tests/logs/rule-custom-2024-03-14-0924/test_suite.log
INFO - xccdf_org.ssgproject.content_rule_accounts_polyinstantiated_tmp
INFO - Script correct.pass.sh using profile (all) OK
INFO - Script line_not_there.fail.sh using profile (all) OK
INFO - Script wrong_mode.fail.sh using profile (all) OK
INFO - Script directory_doesnt_exist.pass.sh using profile (all) OK
INFO - xccdf_org.ssgproject.content_rule_systemd_tmp_mount_enabled
INFO - Script mount_enabled.pass.sh using profile (all) OK
INFO - Script mount_disabled.fail.sh using profile (all) OK
```

**Comment by jan-cerny** [comment]
great!

**Review by Mab879 — APPROVED** [review]

**Pull request merged**